src/libraries/Microsoft.PowerFx.Core/Public/Config/SymbolTable.cs

```diff
@@ -92,16 +93,19 @@ internal void ValidateAccepts(ISymbolSlot slot, FormulaType newType)
         /// <param name="mutable"></param>
         /// <param name="displayName"></param>
         public ISymbolSlot AddVariable(string name, FormulaType type, bool mutable = false, string displayName = null)
-        {
-            if (displayName != null)
-            {
-                // Include parameter so that it's not a breaking change when we enable.
-                // https://github.com/microsoft/Power-Fx/issues/779
-                throw new NotImplementedException("DisplayName support for variables not implemented yet");
-            }
-
+        {
             Inc();
             ValidateName(name);
+            DName displayDName = default;
+
+            if(displayName != null)
+            {
+                displayDName = ValidateName(displayName);
+            }
+            else
+            {
+                displayDName = new DName(name);

```

Comment:
>displayDName = new DName(name); [](http://example.com/codeflow?start=16&length=31)

If there's no displayName, can we leave this null?
Or are we saying now that everything has a displayName, it's just that it may be same as logical name?

Reply:
"we saying now that everything has a displayName", I was planning to go with that approach but realized it can have side effects so moving away from that.